pkg/kubelet/types/pod_update_test.go

```diff
@@ -42,3 +45,131 @@ func TestGetValidatedSources(t *testing.T) {
 	sources, err = GetValidatedSources([]string{"taco"})
 	require.Error(t, err)
 }
+
+func TestGetPodSource(t *testing.T) {
+	cases := []struct {
+		pod      v1.Pod
+		expected string
+	}{
+		{
+			pod:      v1.Pod{},
+			expected: "",
+		},
+		{
+			pod: v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						"kubernetes.io/config.source": "host-ipc-sources",
+					},
+				},
+			},
+			expected: "host-ipc-sources",
+		},
+	}
+	for _, data := range cases {
+		source, err := GetPodSource(&data.pod)
+		if err != nil && source != "" {
```

Comment:
If its an error, you have to assume that it won't return a value or it will be meaningless: don't use `source` if `err != nil`